adam_core/backend/pyoorb.py

```diff
@@ -0,0 +1,326 @@
+try:
+    import pyoorb as oo
+except ImportError:
+    raise ImportError("PYOORB is not installed.")
+
+import os
+import warnings
+from typing import Optional
+
+import numpy as np
+from astropy.time import Time
+
+from ..coordinates.cartesian import CartesianCoordinates
+from ..orbits.orbits import Orbits
+from .backend import Backend
+
+PYOORB_CONFIG = {"dynamical_model": "N", "ephemeris_file": "de430.dat"}
+
+
+class PYOORB(Backend):
+    def __init__(self, **kwargs):
+        # Make sure only the correct kwargs are passed to the constructor
+        allowed_kwargs = PYOORB_CONFIG.keys()
+        for k in kwargs:
+            if k not in allowed_kwargs:
+                warnings.warn(
+                    f"Invalid argument {k} passed to PYOORB backend. Ignoring..."
+                )
+
+        # If an allowed kwarg is missing, add the default
+        for k in allowed_kwargs:
+            if k not in kwargs:
+                kwargs[k] = PYOORB_CONFIG[k]
+
+        self.dynamical_model = kwargs["dynamical_model"]
+        self.ephemeris_file = kwargs["ephemeris_file"]
+
+        env_var = "ADAM_CORE_PYOORB"
+        if env_var in os.environ.keys() and os.environ[env_var] == "True":
+            pass
```

Comment:
What's the purpose behind this part?

Reply:
Oh, I just read further and figured it out.

I still don't really understand, though! So this is to make sure you call oorb_init just once per process tree? Why?

I propose naming the env var something like `ADAM_CORE_PYOORB_INITIALIZED` to be a bit clearer.

Reply:
If you call `oorb_init` twice from the same process it will return an error code 1. So this is an attempt to avoid that.

Reply:
Also, renamed the env var as you suggested.